app/scripts/modules/core/src/cluster/filter/ClusterFilterService.ts

```diff
@@ -334,6 +334,25 @@ export class ClusterFilterService {
     if (filter.includes('cluster:')) {
       const clusterName: string = filter.split('cluster:')[1];
       return serverGroup.cluster === clusterName;
+    }
+
+    if (filter.includes('labels:')) {
+      const labelsStr = filter.split('labels:')[1];
+      const labels = labelsStr.split(',').map(l => l.trim());
+
+      return every(labels, label => {
+        let labelKey: string;
+        let labelValue: string;
+        if (label.includes('=')) {
+          [labelKey, labelValue] = label.split('=');
+        }
+        if (!labelKey || !labelValue) {
+          return false;
+        }
+        return some(serverGroup.labels || {}, (val: string, key: string) => {
+          return labelKey.toLowerCase() === key.toLowerCase() && val.toLowerCase().includes(labelValue.toLowerCase());
```

Comment:
I think these are case-sensitive... do the k8s docs say otherwise?

Reply:
Ah, another case of inconsistency with existing search filters vs. inconsistency with k8s patterns -- I'll remove case insensitivity + partial value match acceptance